tests/ui/pin-ergonomics/pattern-matching-mix-deref-pattern.rs

```diff
@@ -0,0 +1,113 @@
+//@ revisions: pin_ergonomics normal
+//@ edition:2024
+#![cfg_attr(pin_ergonomics, feature(pin_ergonomics))]
+#![feature(deref_patterns)]
+#![allow(incomplete_features)]
+
+// This test verifies that the `pin_ergonomics` feature works well
+// together with the `deref_patterns` feature under the error:
+// "mix of deref patterns and normal constructors".
```

Comment:
This error should be triggered even when `deref_patterns` is not active. Can you remove the `deref_patterns` feature gate on top?